### PR TITLE
feat: verify trusted manufacturers

### PR DESCRIPTION
**Pull request by mmartinv**
- **feat(rendezvous-server)**: verify the device certificate chain
- **feat(owner-onboarding-server)**:  verify the device certificate chain
- **fix(owner-onboarding-server)**: spelling typos

Fixes: #628

### CODE REVIEW

File: rendezvous-server/src/handlers_to0.rs [hunk 1]
```diff
@@ -69,27 +69,6 @@ pub(super) async fn ownersign(
...
-        if !trusted_manufacturer_keys.contains_publickey(&manufacturer_pubkey) {
-            return Err(Error::new(
-                ErrorCode::InvalidOwnershipVoucher,
-                messages::v11::to0::OwnerSign::message_type(),
-                "Ownership voucher manufacturer not trusted",
-            )
-            .into());
-        }
-    }
-
```
Comment (7flying) [inline rendezvous-server/src/handlers_to0.rs:92]: This part is actually OK and we shouldn't delete it.
`to0d.OV.header[4]` holds the manufacturer's public key, as specified at <https://fidoalliance.org/specs/FDO/FIDO-Device-Onboard-PS-v1.1-20220419/FIDO-Device-Onboard-PS-v1.1-20220419.html#OwnershipVoucher>.
The manufacturer server stores its public cert there ([code](https://github.com/fdo-rs/fido-device-onboard-rs/blob/main/manufacturing-server/src/handlers/di.rs#L133)), so in this part we are just doing an allowlist check.

Comment (7flying) [reply rendezvous-server/src/handlers_to0.rs:92]: this also comes [from this](https://fidoalliance.org/specs/FDO/FIDO-Device-Onboard-PS-v1.1-20220419/FIDO-Device-Onboard-PS-v1.1-20220419.html#TO0:~:text=It%20is%20preferred,such%20a%20trust.) :
> It is preferred that the Rendezvous Server has a basis on which to trust at least one public key within the Ownership Voucher. For example, the manufacturer who ran the DI protocol to configure the Device, thereby choosing the Rendezvous Server, may register public keys with the Rendezvous Server to establish such a trust.

File: rendezvous-server/src/handlers_to0.rs [hunk 2]
```diff
@@ -157,8 +136,15 @@ pub(super) async fn ownersign(
         }
         Some(v) => v,
     };
-    //let device_pubkey = match device_cert_chain.verify_from_x5bag(&user_data.trusted_device_keys) {
-    let device_pubkey = match device_cert_chain.insecure_verify_without_root_verification() {
+
+    let device_pubkey_verification =
+        if let Some(trusted_manufacturer_certs) = &user_data.trusted_manufacturer_keys {
+            device_cert_chain.verify_from_x5bag(trusted_manufacturer_certs)
```
Comment (7flying) [inline rendezvous-server/src/handlers_to0.rs:142]: So, I think that the issue here is whether we can trust any certificate as root or not and whether the rendezvous has the abilities to verify root certificates...

Comment (7flying) [reply rendezvous-server/src/handlers_to0.rs:142]: what was the error that we were getting before changing the ca cert in the tests?

Comment (mmartinv) [reply rendezvous-server/src/handlers_to0.rs]: > what was the error that we were getting before changing the ca cert in the tests?

Honestly, I don't remember, I would need to check the conformance tests again. 
Either way, we need the servers to be able to validate the device certificate chain by using a list of trusted CA certs (apart from verifying that the chain is correct).

**Review by 7flying — APPROVED** [review]

**Pull request merged**